### PR TITLE
Fix 16888: create gpkg layer does not allow to add Z/M dimension

### PR DESCRIPTION
**Pull request by ghost**
Fix issue: https://issues.qgis.org/issues/16888

### CODE REVIEW

**Comment by nyalldawson** [comment]
Looks like there's some unrelated commits in this PR - can you clean these up or clarify if they're required?

(Incidentally - please give a description in the PR title/description about what the related bug is - e.g. "Fix 16888: create gpkg layer does not allow to add Z/M dimension". It just makes following the PR queue a bit easier then dealing with raw issue numbers alone)

File: src/gui/qgsnewgeopackagelayerdialog.cpp [hunk 1]
```diff
@@ -349,6 +349,13 @@ bool QgsNewGeoPackageLayerDialog::apply()
   OGRwkbGeometryType wkbType = static_cast<OGRwkbGeometryType>
                                ( mGeometryTypeBox->currentData( Qt::UserRole ).toInt() );
 
+  // z-coordinate & m-value.
+  if (mGeometryWithZCheckBox->isChecked())
```
Comment (nyalldawson) [inline src/gui/qgsnewgeopackagelayerdialog.cpp]: I'd prefer to see these split into 2 separate checkboxes. There's often times when you'd want Z or M without wanting both dimensions.

**Comment by m-kuhn** [comment]
To cleanup the commit history, do this:

```
git checkout master
# git pull from origin/upstream/qgis whatever you called it
# Create a new branch fixGpkgZm (based on master)
git checkout -b fixGpkgZm
# Apply only the missing commit
git cherry-pick c9f66dc
# Push the new branch
git push -u fixGpkgZm
```

Whenever you start working on a new issue/feature, start a new branch from the current master branch, this will keep things well separated and make it easier to review and merge commits.

**Comment by nouanda** [comment]
@m-kuhn Thanks for the tip. I have created a new pull request with cleaned commits.
PR: https://github.com/qgis/QGIS/pull/5060

**Pull request closed without merging**